plugins/catalyst/python/pyproject.toml

```diff
@@ -0,0 +1,298 @@
+  'error',
+  'ignore:\s.*Pyarrow.*:DeprecationWarning:',
+  'ignore:.*datetime\.datetime\.utcfromtimestamp.*:DeprecationWarning:',
+  'ignore:.*jax\.interpreters\.mlir.*:DeprecationWarning:',
+]
+log_cli_level = "INFO"
+testpaths = ["test"]
+
+[tool.coverage]
+run.source = ["mqt.catalyst"]
+run.omit = []
+report.exclude_also = [
+  '\.\.\.',
+  'if TYPE_CHECKING:',
+  'raise AssertionError',
+  'raise NotImplementedError',
+  'def __dir__()',  # Ignore __dir__ method that exists mainly for better IDE support
+  '@overload'  # Overloads are only for static typing
+]
+
+[tool.mypy]
+files = ["src", "test"]
+mypy_path = ["$MYPY_CONFIG_FILE_DIR/src"]
+python_version = "3.9"
+warn_unused_configs = true
+enable_error_code = ["ignore-without-code", "redundant-expr", "truthy-bool"]
+strict = true
+disallow_untyped_defs = false
+explicit_package_bases = true
+warn_unreachable = true
+
+[[tool.mypy.overrides]]
+module = []
+ignore_missing_imports = true
+
+[tool.ruff]
+line-length = 120
+namespace-packages = ["mqt"]
+preview = true
+unsafe-fixes = true
+
+[tool.ruff.lint]
+select = ["ALL"]
+ignore = [
+    "C90",     # <...> too complex
+    "COM812",  # Conflicts with formatter
+    "CPY001",  # Missing copyright notice at top of file
+    "ISC001",  # Conflicts with formatter
+    "PLR09",   # Too many <...>
+    "PLR2004", # Magic value used in comparison
+    "PLC0415", # Import should be at top of file
+    "S101",    # Use of assert detected
+    "S404",    # `subprocess` module is possibly insecure
+    "TID252"   # Prefer absolute imports over relative imports from parent modules
+]
+# todo: can the module be used from mqt-core?
+typing-modules = ["mqt.core._compat.typing"]
+isort.required-imports = ["from __future__ import annotations"]
+
+[tool.ruff.lint.flake8-tidy-imports.banned-api]
+"typing.Callable".msg = "Use collections.abc.Callable instead."
+"typing.Iterator".msg = "Use collections.abc.Iterator instead."
+"typing.Mapping".msg = "Use collections.abc.Mapping instead."
+"typing.Sequence".msg = "Use collections.abc.Sequence instead."
+"typing.Set".msg = "Use collections.abc.Set instead."
+"typing.assert_never".msg = "Use mqt.core._compat.typing.assert_never instead."
+
+[tool.ruff.lint.per-file-ignores]
+# todo: remove the following lines and fix the python files themselves
+"src/mqt/catalyst/__init__.py" = ["ALL"]
+"test/**" = ["ALL"]
+
+[tool.ruff.lint.pydocstyle]
+convention = "google"
+
+[tool.typos]
+default.extend-ignore-re = [
+  "(?Rm)^.*(#|//)\\s*spellchecker:disable-line$",  # ignore line
+  "(?s)(#|//)\\s*spellchecker:off.*?\\n\\s*(#|//)\\s*spellchecker:on" # ignore block
+]
+
+[tool.typos.default.extend-words]
+wille = "wille"
+anc = "anc"
+mch = "mch"
+ket = "ket"
+
+[tool.repo-review]
+ignore = ["GH200"]
+
+[tool.cibuildwheel]
+build = "cp3*"
+skip = "*-musllinux_*"
+archs = "auto64"
+test-groups = ["test"]
+test-command = "pytest {project}/test"
+build-frontend = "build[uv]"
+enable = ["cpython-freethreading"]
```

Comment:
pretty sure catalyst does not provide free threading builds. so this won't work and can be removed.